linux/src/async_socket_linux.c

```diff
@@ -274,6 +282,12 @@ static void internal_close(ASYNC_SOCKET_HANDLE async_socket)
         completion_port_remove(async_socket->completion_port, async_socket->socket_handle);
     }
 
+    // Codes_SRS_ASYNC_SOCKET_LINUX_11_103: [ async_socket_close shall wait for the event_complete_callback to complete ]
+    while ((value = interlocked_add(&async_socket->event_complete_executing, 0)) != 0)
+    {
+        (void)wait_on_address(&async_socket->event_complete_executing, value, UINT32_MAX);
```

Comment:
>UINT32_MAX [](http://example.com/codeflow?start=78&length=10)

`while` (line 286) and `UINT32_MAX` here really prepare us for eternity.

How about having a timeout of 1 second and Log something every time `wait_on_address` doesn't get it signalled? That way it is more descriptive of what happened.

Sort of here and everywhere as I see the same pattern used in other places too.

Reply:
I see, but the issue would be we could get 1000s of logs on a situation that is not a problem and it may happens often.